src/include/catalog/catalog_accessor.h

```diff
@@ -380,6 +381,19 @@ class EXPORT CatalogAccessor {
    */
   common::ManagedPointer<transaction::TransactionContext> GetTxn() const { return txn_; }
 
+  /**
+   * Registers a temporary table in this accessor
+   * @param table_oid The temporary oid of this table
+   * @param table The temp table being registered
+   */
+  void RegisterTempTable(table_oid_t table_oid, common::ManagedPointer<storage::SqlTable> table);
+
+  /**
+   * Allocates and returns a new temporary oid. These oids are only valid for the lifetime of this accessor
```

Comment:
How does this work with caching? What happens if another query comes along and tries to use the cached plans but for some reason the oids don't match? Is that possible? Are there other implications for query caching with this design?

Reply:
It should still work. All uses of this are restricted to the backend of the execution engine so cached plans should behave the same way. I'll make note of this.